src/App.tsx

```diff
@@ -14,11 +15,91 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [goods] = useState(goodsFromServer);
+  const [isStart, setIsStart] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const startButton = () => {
+    setIsStart(() => true);
+  };
+
+  const reverseButton = () => {
+    setIsReverse(prev => !prev);
+  };
+
+  const resetButton = () => {
+    setIsReverse(() => false);
+    setSortBy(() => '');
+  };
+
+  const sortButton = (sortStr: string) => {
+    setSortBy(() => sortStr);
+  };
+
+  const prepGoods: string[] = sortBy !== ''
+    ? [...goods].sort((good1, good2) => {
+      switch (sortBy) {
+        case 'alphab': return good1.localeCompare(good2);
+        case 'length': return good1.length - good2.length;
+        default: return 0;
+      }
+    })
+    : [...goods];
```

Comment:
Just add some logic to default case

Reply:
sudenly 0 is logic(array did not sort)

Reply:
You don't need ternary operator here, just pass correct callback in default case